compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -461,8 +461,8 @@ fn run_compiler(
         // Linking is done outside the `compiler.enter()` so that the
         // `GlobalCtxt` within `Queries` can be freed as early as possible.
         if let Some(linker) = linker {
-            let _timer = sess.timer("link");
-            linker.link(sess, codegen_backend)?
+            let _timer = sess.timer("waiting for linking");
```

Comment:
Can we keep a timer for the entire time linking takes?

Reply:
We still have `link_crate` which covers the codegen backend part of linking.